x/wasm/internal/keeper/authorization_policy.go

```diff
@@ -0,0 +1,42 @@
+package keeper
+
+import (
+	"github.com/CosmWasm/wasmd/x/wasm/internal/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type AuthorizationPolicy interface {
+	CanCreateCode(c types.AccessConfig, creator sdk.AccAddress) bool
+	CanInstantiateContract(c types.AccessConfig, actor sdk.AccAddress) bool
+	CanModifyContract(c types.AccessConfig, actor sdk.AccAddress) bool
+}
+
+type DefaultAuthorizationPolicy struct {
+}
+
+func (p DefaultAuthorizationPolicy) CanCreateCode(config types.AccessConfig, actor sdk.AccAddress) bool {
+	return config.Allowed(actor)
+}
+
+func (p DefaultAuthorizationPolicy) CanInstantiateContract(config types.AccessConfig, actor sdk.AccAddress) bool {
+	return config.Allowed(actor)
+}
+
+func (p DefaultAuthorizationPolicy) CanModifyContract(config types.AccessConfig, actor sdk.AccAddress) bool {
```

Comment:
Since we have 3 calls here, maybe we can just use a different sig for `CanModifyContract` to behave with `admin = nil | Address` as now. And the create/instantiate using the new AccessConfig you created.

Reply:
Actually I had this before before I unified to `AccessConfig`.